src/lib/providers/Web3Provider.tsx

```diff
@@ -0,0 +1,39 @@
+import { providers } from "ethers";
+import { FC, createContext, ReactNode } from "react";
+import { USER_ADDRESS } from "../constants/addresses";
+
+interface Web3ProviderProps {
+  provider?: providers.Web3Provider;
+  children?: ReactNode;
+}
+
+const INFURA_MAINNET =
+  "https://mainnet.infura.io/v3/77c3d733140f4c12a77699e24cb30c27";
+
+export const Web3Context = createContext({
+  address: undefined as string | undefined,
+  provider: undefined as providers.Web3Provider | undefined,
+});
+
+const Web3Provider: FC<Web3ProviderProps> = ({
+  provider,
+  children,
+}: Web3ProviderProps) => {
```

Comment:
Do we still need this Provider?

Reply:
It's copied from my repo - I kept it so you could still access `{ account, provider }` through `useWeb3()` instead of prop drilling. Could alternatively store these in Redux.